go/cmd/betabot/main.go

```diff
@@ -58,13 +59,23 @@ type Bot struct {
 }
 
 type Conversation struct {
+	sync.Mutex
+	ConversationJSON
+}
+
+type ConversationJSON struct {
```

Comment:
can't you just put the sync.Mutex inside the Conversation?

Reply:
Same as previous, no because I need an object without locker for {M,Unm}arshal due to go vet.

Reply:
ok, so switch to a map of mutex

Reply:
for your information, seeing this PR making overkill change and increasing complexity just gives me the motivation to setup gorm right know

so please just focus on fixing the race (which is a real problem) instead of trying to make a too much smart store

Reply:
> 
> 
> for your information, seeing this PR making overkill change and increasing complexity just gives me the motivation to setup gorm right know
> 
> so please just focus on fixing the race (which is a real problem) instead of trying to make a too much smart store

I'll do, but the store is still dumb AF, except there is 2, one with lockers for betabot usage and one without for json marshaling.